components/brave_shields/resources/panel/state/hooks.ts

```diff
@@ -43,3 +43,19 @@ export function useSiteSettingsData () {
 
   return { siteSettings, getSiteSettings }
 }
+
+export function useFavIconUrl (url?: string) {
+  const [favIconUrl, setfavIconUrl] = React.useState<string>()
+
+  React.useEffect(() => {
```

Comment:
I'm not sure why we're using `useState` and `useEffect` when all we want to return is a `faviconUrl` based on the incoming `url`. Can't we just use `useMemo` to avoid an excessive state change and re-render after the useEffect runs?

```ts
return React.useMemo(() => {
    if (!url) return
    const faviconUrl = new URL('chrome://favicon2/')
    faviconUrl.searchParams.set('size', '16')
    faviconUrl.searchParams.set('scale_factor', '2x')
    faviconUrl.searchParams.set('show_fallback_monogram', '')
    faviconUrl.searchParams.set('page_url', `https://${url}`)
    return faviconUrl.toString()
}, [url])
```

Reply:
getting this from backend now